File: LC_2024_Maximize_the_Confusion_of_an_Exam.py
class Solution:
    def maxConsecutiveAnswers(self, answerKey: str, k: int) -> int:
        l,ans=0,0
        fcount,tcount=0,0
        for r in range(len(answerKey)):
            if answerKey[r]=='F':
                fcount+=1
            if answerKey[r]=='T':
                tcount+=1
            while min(fcount,tcount)>k:
                if answerKey[l]=='F':
                    fcount-=1
                if answerKey[l]=='T':
                    tcount-=1
                l+=1
            ans=max(ans,r-l+1)
        return ans

File: LC_2024_Maximize the Confusion of an Exam.java
class Solution {
    public int maxConsecutiveAnswers(String answerKey, int k) {
        int l=0,temp=0,ans=0;
        int fcount=0,tcount=0;
        for(int r=0;r<answerKey.length();r++){
            if(answerKey.charAt(r)=='F'){
                fcount++;
            }
            if(answerKey.charAt(r)=='T'){
                tcount++;
            }
            while(Math.min(fcount,tcount)>k){
                if(answerKey.charAt(l)=='F'){
                    fcount--;
                }
                if(answerKey.charAt(l)=='T'){
                    tcount--;
                }
                l++;
            }
            ans=Math.max(ans,r-l+1);
        }
        return ans;
    }
}